app/src/main/java/com/randomappsinc/studentpicker/common/VoiceRecognizerDialogFragment.java

```diff
@@ -0,0 +1,161 @@
+package com.randomappsinc.studentpicker.common;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.speech.RecognitionListener;
+import android.speech.RecognizerIntent;
+import android.speech.SpeechRecognizer;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.DialogFragment;
+
+import com.joanzapata.iconify.widget.IconTextView;
+import com.randomappsinc.studentpicker.R;
+
+import java.util.ArrayList;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+public class VoiceRecognizerDialogFragment extends DialogFragment implements RecognitionListener {
```

Comment:
We should use MaterialDialog instead. Its material design is backported and it abstracts a lot of the fragment management away. It also has a powerful set of APIs. We can just place a custom view inside of it and bind to it.

Also, we should create SpeechToTextManager and have it manage the UI (fragment) and speech recognition separately. We don't want to couple them together because it limits what we can do with them, makes the code less reusable, and makes our code prone to becoming monolithic (1 giant class). We can even wrap the MaterialDialog into its own component like SpeechToTextDialog and expose APIs to mutate the UI.

Reply:
So I have ta make SpeechToTextManager class that handles Voice Recognizing, also has showDialog function and of course using MaterialDialog with custom view.
Is that what you mean by saying "wrap the MaterialDialog into its own component" ?

Reply:
Wrapping the MaterialDialog in its own component is creating yet another class just for the dialog UI. That's kind of a bonus; you don't need to do it.